src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -0,0 +1,289 @@
+    if xr_engine == "h5netcdf":
+        xr_open_args["engine"] = "h5netcdf"
+        xr_open_args["lock"] = False
+    else:
+        # Zarr arguments
+        xr_open_args["engine"] = "zarr"
+        xr_open_args["consolidated"] = consolidated
+
+    # Additional arguments when dealing with a reference file.
+    if reference:
+        xr_open_args["consolidated"] = False
+        xr_open_args["backend_kwargs"] = {"consolidated": False}
+
+    ds = xarray.open_dataset(file_handler, **xr_open_args)
+
+    if cache_client:
+        # Serialize the dataset to bytes using pickle
+        data_bytes = pickle.dumps(ds)
+        cache_client.set(cache_key, data_bytes)
+
+    return ds
+
+
+def _arrange_dims(da: xarray.DataArray) -> xarray.DataArray:
+    """Arrange coordinates and time dimensions.
+
+    An rioxarray.exceptions.InvalidDimensionOrder error is raised if the coordinates are not in the correct order time, y, and x.
+    See: https://github.com/corteva/rioxarray/discussions/674
+
+    We conform to using x and y as the spatial dimension names..
+
+    """
+    if "x" not in da.dims and "y" not in da.dims:
+        latitude_var_name = next(
+            x for x in ["lat", "latitude", "LAT", "LATITUDE", "Lat"] if x in da.dims
+        )
+        longitude_var_name = next(
+            x for x in ["lon", "longitude", "LON", "LONGITUDE", "Lon"] if x in da.dims
+        )
+        da = da.rename({latitude_var_name: "y", longitude_var_name: "x"})
+
+    if "TIME" in da.dims:
+        da = da.rename({"TIME": "time"})
+
+    if _dims := [d for d in da.dims if d not in ["x", "y"]]:
+        da = da.transpose(*_dims, "y", "x")
+    else:
+        da = da.transpose("y", "x")
+
+    # If min/max values are stored in `valid_range` we add them in `valid_min/valid_max`
+    vmin, vmax = da.attrs.get("valid_min"), da.attrs.get("valid_max")
+    if "valid_range" in da.attrs and not (vmin is not None and vmax is not None):
+        valid_range = da.attrs.get("valid_range")
+        da.attrs.update({"valid_min": valid_range[0], "valid_max": valid_range[1]})
+
+    return da
+
+
+def get_variable(
+    ds: xarray.Dataset,
+    variable: str,
+    datetime: Optional[str] = None,
+    drop_dim: Optional[str] = None,
+) -> xarray.DataArray:
+    """Get Xarray variable as DataArray."""
+    da = ds[variable]
+
+    if drop_dim:
+        dim_to_drop, dim_val = drop_dim.split("=")
+        da = da.sel({dim_to_drop: dim_val}).drop_vars(dim_to_drop)
+
+    da = _arrange_dims(da)
+
+    # Make sure we have a valid CRS
+    crs = da.rio.crs or "epsg:4326"
+    da = da.rio.write_crs(crs)
+
+    if crs == "epsg:4326" and (da.x > 180).any():
+        # Adjust the longitude coordinates to the -180 to 180 range
+        da = da.assign_coords(x=(da.x + 180) % 360 - 180)
+
+        # Sort the dataset by the updated longitude coordinates
+        da = da.sortby(da.x)
+
+    # TODO: Technically we don't have to select the first time, rio-tiler should handle 3D dataset
```

Comment:
☝️ we might enable all the `times` values, and not select the first value by default